File: app/src/main/java/de/metzgore/beansplan/util/di/components/AppComponent.java
package de.metzgore.beansplan.util.di.components;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import de.metzgore.beansplan.BeansPlanApp;
import de.metzgore.beansplan.util.di.modules.ContextModule;
import de.metzgore.beansplan.util.di.modules.DailyScheduleDaoModule;
import de.metzgore.beansplan.util.di.modules.MainActivityComponent;
import de.metzgore.beansplan.util.di.modules.RepositoryModule;

import javax.inject.Singleton;

@Singleton
@Component(modules = {AndroidInjectionModule.class,
        RepositoryModule.class,
        DailyScheduleDaoModule.class,
        ContextModule.class,
        MainActivityComponent.class})
public interface AppComponent {

    void inject(BeansPlanApp app);
}

File: app/src/main/java/de/metzgore/beansplan/util/di/components/AppComponent.kt
package de.metzgore.beansplan.util.di.components

import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.support.AndroidSupportInjectionModule
import de.metzgore.beansplan.BeansPlanApp
import de.metzgore.beansplan.util.di.modules.ClockModule
import de.metzgore.beansplan.util.di.modules.ContextModule
import de.metzgore.beansplan.util.di.modules.RepositoryModule
import de.metzgore.beansplan.util.di.modules.ScheduleDaoModule
import de.metzgore.beansplan.util.di.modules.SettingsModule
import javax.inject.Singleton

@Singleton
@Component(modules = [(AndroidInjectionModule::class), (AndroidSupportInjectionModule::class),
    (ClockModule::class), (RepositoryModule::class), (ScheduleDaoModule::class),
    (ContextModule::class), (MainActivityComponent::class), (WeeklyScheduleFragmentComponent::class),
    (DailyScheduleFragmentComponent::class), (RemindersFragmentComponent::class),
    (ReminderDeletionDialogFragmentComponent::class), (ReminderTimePickerDialogFragmentComponent::class),
    (ReminderDeletionOrUpdateDialogFragmentComponent::class), (NotificationPublisherComponent::class),
    (ReminderBootReceiverComponent::class), (SettingsModule::class), (SettingsFragmentComponent::class),
    (SettingsActivityComponent::class)])
interface AppComponent {
    fun inject(app: BeansPlanApp)
}
